packages/demo/usecase-demo/src/components/useCarouselLogic.ts

```diff
@@ -0,0 +1,15 @@
+export function OnScrollLeft(carousel) {
+  carousel.current.scrollBy({
+    top: 0,
+    left: -carousel.current.clientWidth,
+    behavior: 'smooth',
+  });
+}
+
+export function OnScrollRight(carousel) {
+  carousel.current.scrollBy({
+    top: 0,
+    left: carousel.current.clientWidth,
+    behavior: 'smooth',
+  });
+}
```

Comment:
- functions should always start with lowercase letter: "OnScrollLeft" → "onScrollLeft"


- This doesn't look like a React hook, so the filename should not start with the keyword `use`. "useCarouselLogic.ts" → "carouselLogic.ts"

If it were a hook:
```
export function useCarouselLogic() {
  function OnScrollLeft(carousel) {
    // ...
  }

  function OnScrollRight(carousel) {
    // ...
  }

  return {
    OnScrollLeft,
    OnScrollRight,
  };
}

```


- This is not a component so I would move this file to our "utils" folder.


- Maybe name these functions just "scrollLeft()" and "scrollRight()"?


- Missing TypeScript type for `carousel` argument. Maybe: `React.RefObject<HTMLDivElement>`